src/main/java/net/minecraftforge/common/DimensionManager.java

```diff
@@ -49,111 +47,82 @@
 import net.minecraft.world.WorldServer;
 import net.minecraft.world.WorldServerMulti;
 import net.minecraft.world.storage.ISaveHandler;
+import net.minecraftforge.common.Dimension;
 import net.minecraftforge.event.world.WorldEvent;
-import net.minecraftforge.fml.common.FMLCommonHandler;
 import net.minecraftforge.fml.common.FMLLog;
+import net.minecraftforge.fml.common.registry.ForgeRegistries;
 
 import javax.annotation.Nullable;
 
 public class DimensionManager
 {
-    private static class Dimension
-    {
-        private final DimensionType type;
-        private int ticksWaited;
-        private Dimension(DimensionType type)
-        {
-            this.type = type;
-            this.ticksWaited = 0;
-        }
-    }
-
-    private static Hashtable<Integer, WorldServer> worlds = new Hashtable<Integer, WorldServer>();
-    private static boolean hasInit = false;
-    private static Hashtable<Integer, Dimension> dimensions = new Hashtable<Integer, Dimension>();
-    private static IntArrayList unloadQueue = new IntArrayList();
-    private static BitSet dimensionMap = new BitSet(Long.SIZE << 4);
+    private static Hashtable<ResourceLocation, WorldServer> worlds = new Hashtable<ResourceLocation, WorldServer>();
+    private static Set<ResourceLocation> activeDimensions = new HashSet<ResourceLocation>();
+    private static List<ResourceLocation> unloadQueue = new ArrayList<ResourceLocation>();
     private static ConcurrentMap<World, World> weakWorldMap = new MapMaker().weakKeys().weakValues().<World,World>makeMap();
     private static Multiset<Integer> leakedWorlds = HashMultiset.create();
 
     /**
      * Returns a list of dimensions associated with this DimensionType.
      */
-    public static int[] getDimensions(DimensionType type)
-    {
-        int[] ret = new int[dimensions.size()];
-        int x = 0;
-        for (Map.Entry<Integer, Dimension> ent : dimensions.entrySet())
-        {
-            if (ent.getValue().type == type)
-            {
-                ret[x++] = ent.getKey();
-            }
-        }
-
-        return Arrays.copyOf(ret, x);
-    }
-
-    public static void init()
+    public static ResourceLocation[] getDimensions(DimensionType type)	//Could return Dimension[] since this contains name info
```

Comment:
Should this be changed to return some kind of more general collection like https://github.com/MinecraftForge/MinecraftForge/pull/4545 since this will be a breaking change? Or should I follow that PR and just add another method to return a generic collection?